resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DbReader.scala

```diff
@@ -60,6 +61,23 @@ object DbReader {
           )"""
       .query[Runtime]
 
+  val deletingRuntimeQuery =
+    sql"""SELECT DISTINCT c1.id, cloudContext, c1.cloudProvider, runtimeName, rt.cloudService, c1.status, rt.zone, rt.region
+          FROM CLUSTER AS c1
+          INNER JOIN RUNTIME_CONFIG AS rt ON c1.runtimeConfigId = rt.id
+          WHERE
+            c1.status = "Deleting" AND
+            c1.dateAccessed < now() - INTERVAL 1 HOUR AND
+            NOT EXISTS (
```

Comment:
Trying to understand why we need the `AND NOT EXISTS …` clause here. It seems unnecessary, since there is a unique constraint on `(cloudContext, name, destroyedDate)`?

Reply:
From Qi's comment above, it is necessary precaution as AOU reuses names

Reply:
I think that would only be necessary for `Deleted` runtimes though, otherwise there is a DB constraint.

Reply:
since the runtime is `Deleting`...I don't think `destroyedDate` will be defined yet... and (cloudContext, name) is not guranteed to be unique

Reply:
Yeah we only set `destroyedDate` when the status is `Deleted`… which means the constraint on `(cloudContext, name, destroyedDate)` only allows reusing names when at most 1 is non-deleted. (So you can't have a Deleting and Running runtime with the same name at the same time.)

I'm fine with leaving it as a sanity check… I was also wondering why the status check is not complete though. For example what if there were 2 runtimes:
```
name | status
-----
myRuntimeName | Deleted
myRuntimeName | Stopping
```

Would the `deletedRuntimeSql` mistakenly flag `myRuntimeName` and delete it in GCP?

Reply:
I think the `NOT EXISTS` exclude `Stopped` as well (both `Stopped` and `Running` are "active" statuses basically)..

but yea, it's tricky...think we did this only becuz AOU reuses names...I'll bring it up to see if we can use unique names going forward for AOU

Reply:
created https://precisionmedicineinitiative.atlassian.net/browse/RW-10776